### PR TITLE
adding solution and tests for challenge 6

### PR DESCRIPTION
**Pull request by jacobgarner**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -94,27 +93,66 @@ describe("/api/articles/:article_id", () => {
...
+});
+
+describe("/api/articles/:article_id/comments", () => {
+  it("GET responds with status code 200", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.comments.length).toBe(2);
+        expect(response.body.comments[0].article_id).toBe(3);
```
Comment (Hard-Luck) [inline __tests__/app.test.js]: you need to check for each comment

File: __tests__/app.test.js [hunk 2]
```diff
@@ -94,27 +93,66 @@ describe("/api/articles/:article_id", () => {
...
+        expect(response.body.comments[0]).toMatchObject({
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          article_id: expect.any(Number),
```
Comment (Hard-Luck) [inline __tests__/app.test.js]: You know the article id

File: __tests__/app.test.js [hunk 3]
```diff
@@ -94,27 +93,66 @@ describe("/api/articles/:article_id", () => {
...
+          author: expect.any(String),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
```
Comment (Hard-Luck) [inline __tests__/app.test.js:146]: You need to check the order of all the comments, i would recommend picking an article with more comments so you can be confident your sorting is working, use something like jest-sorted to check the order

File: models/topicsModel.js [hunk 4]
```diff
@@ -40,3 +39,10 @@ exports.getAllArticles = () => {
       return articles.rows;
     });
 };
+
+exports.getCommentsByArticleId = (article_id) =>{
+    return db.query(`SELECT * FROM comments
```
Comment (Hard-Luck) [inline models/topicsModel.js:44]: This wont ensure the order of the comments, think about what clauses you could add to your command

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Really good start, some key things missing, take a look at my comments. :)

File: __tests__/app.test.js [hunk 5]
```diff
@@ -61,7 +60,7 @@ describe("/api/articles/:article_id", () => {
       .get("/api/articles/2")
```
Comment (nc-mick-fay) [inline __tests__/app.test.js:60]: these tests can be chained together, as you can follow your `expect(200)` with a `.then((response) => {`

File: __tests__/app.test.js [hunk 6]
```diff
@@ -94,27 +93,71 @@ describe("/api/articles/:article_id", () => {
...
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((response) => {
+        console.log(response.body.comments)
+        expect(response.body.comments).toBeSortedBy("created_at", {
+          descending: true,
+        })
+        expect(response.body.comments.length).toBe(11);
+        expect(response.body.comments[0].article_id).toBe(1);
```
Comment (nc-mick-fay) [inline __tests__/app.test.js]: You want your article_id to be 1 for every single comment, not just the first one - move this down into your forEach loop s out applies to each comment

File: __tests__/app.test.js [hunk 7]
```diff
@@ -94,27 +93,71 @@ describe("/api/articles/:article_id", () => {
...
+  });
+  it("Responds with 'does not exist' if valid article ID provided that does not exist", () => {
+    return request(app)
+      .get("/api/articles/345/comments")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.err).toBe("Article does not exist");
+      });
+  });
+});
```
Comment (nc-mick-fay) [inline __tests__/app.test.js:170]: What does your endpoint return when an article that has no comments is requested?

**Review by nc-mick-fay — CHANGES_REQUESTED** [review]
Hi Jacob, really nice additions/changes to account for previous reviews. 

We are still missing a test for if you make a request for an article that has no comments. This should return a 200 with an empty array.

Once this is added you'll be good to merge!

**Review by nc-mick-fay — APPROVED** [review]
Looks good, ready to merge!

**Pull request merged**